src/main/groovy/org/pih/warehouse/inventory/ProductInventorySnapshotTaker.groovy

```diff
@@ -0,0 +1,69 @@
+package org.pih.warehouse.inventory
+
+import grails.validation.ValidationException
+import org.springframework.stereotype.Component
+
+import org.pih.warehouse.api.AvailableItem
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.product.Product
+
+/**
+ * Responsible for creating and persisting product inventory transactions. We call this a "snapshot taker" because
+ * product inventory transactions saved via this class are functionally a copy of what the quantity on hand
+ * (determined by product availability) was for each [bin location + lot number] combo of the product at the time.
+ * Importantly, transactions created via this class should NOT result in any quantity changes/adjustments.
+ */
+@Component
+class ProductInventorySnapshotTaker {
```

Comment:
@jmiranda added you as a reviewer primarily to check out this file. I figure that as we move towards the "product inventory as snapshot" approach, we'll want this code to be in a class like this so it's easier to integrate into other places while keeping the logic the same.